gno.land/pkg/integration/gnoland.go

```diff
@@ -0,0 +1,359 @@
+package integration
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gnolang/gno/gno.land/pkg/gnoland"
+	vmm "github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/config"
+	"github.com/gnolang/gno/tm2/pkg/bft/node"
+	"github.com/gnolang/gno/tm2/pkg/bft/privval"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/db"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+type IntegrationConfig struct {
+	SkipFailingGenesisTxs bool
+	SkipStart             bool
+	GenesisBalancesFile   string
+	GenesisTxsFile        string
+	ChainID               string
+	GenesisRemote         string
+	RootDir               string
+	GenesisMaxVMCycles    int64
+	Config                string
+}
+
+// NOTE: This is a copy of gnoland actual flags.
+// XXX: A lot this make no sense for integration.
+func (c *IntegrationConfig) RegisterFlags(fs *flag.FlagSet) {
```

Comment:
Unfortunately, I can't import `cmd/gnoland` because it is forbidden. I don't think we'll keep it like that anyway. I ultimately plan to move this file into the `gnoland` package directly as an `InMemory` node so we can centralize everything. However, I'm saving that for another iteration as it will also require reworking `cmd/gnoland` and `pkg/gnoland` for better flexibility.